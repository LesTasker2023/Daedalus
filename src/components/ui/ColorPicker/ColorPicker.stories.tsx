import type { Meta, StoryObj } from '@storybook/nextjs';
import { ColorPicker } from './ColorPicker';

const meta: Meta<typeof ColorPicker> = {
  title: 'Components/ColorPicker',
  component: ColorPicker,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
