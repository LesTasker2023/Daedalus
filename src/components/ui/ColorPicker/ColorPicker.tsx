'use client';

import { Palette } from 'lucide-react';
import { useCallback, useEffect, useState } from 'react';
import { Button } from '../Button';

interface ColorPickerProps {
  defaultColor?: string;
  onColorChange?: (color: string) => void;
  className?: string;
}

const PRESET_COLORS = [
  '#18181b', // Default Zinc - excellent contrast
  '#dc2626', // Red - good contrast
  '#ea580c', // Orange - good contrast
  '#ca8a04', // Yellow - moderate contrast
  '#16a34a', // Green - good contrast
  '#0891b2', // Cyan - good contrast
  '#2563eb', // Blue - excellent contrast
  '#7c3aed', // Violet - good contrast
  '#c026d3', // Fuchsia - good contrast
  '#e11d48', // Rose - good contrast
  '#1f2937', // Dark Gray - excellent contrast
  '#4f46e5', // Indigo - excellent contrast
];

function ColorPicker({
  defaultColor = '#18181b',
  onColorChange,
  className = '',
}: ColorPickerProps) {
  const [isOpen, setIsOpen] = useState(false);
  const [selectedColor, setSelectedColor] = useState(defaultColor);
  const [customColor, setCustomColor] = useState(defaultColor);
  const [mounted, setMounted] = useState(false);

  // Function to update dark mode CSS variables dynamically
  const updateDarkModeColors = (color: string, foregroundColor: string) => {
    // Check if a dynamic dark mode style tag already exists
    let styleTag = document.getElementById('dynamic-dark-colors');

    if (!styleTag) {
      styleTag = document.createElement('style');
      styleTag.id = 'dynamic-dark-colors';
      document.head.appendChild(styleTag);
    }

    // Create CSS rule for dark mode brand colors
    styleTag.textContent = `
      .dark {
        --brand-primary: ${color} !important;
        --brand-primary-foreground: ${foregroundColor} !important;
      }
    `;
  };

  // Function to apply color to document and localStorage
  const applyColorToDocument = useCallback((color: string) => {
    // Update CSS variables for light mode
    document.documentElement.style.setProperty('--brand-primary', color);

    // Calculate proper contrast for foreground
    const rgb = hexToRgb(color);
    if (rgb) {
      const luminance = getLuminance(rgb);
      const foregroundColor = luminance > 0.5 ? '#000000' : '#ffffff';
      document.documentElement.style.setProperty('--brand-primary-foreground', foregroundColor);

      // Update dark mode versions
      updateDarkModeColors(color, foregroundColor);
    }

    // Save to localStorage for persistence
    localStorage.setItem('brand-color', color);
  }, []);

  // Initialize colors from localStorage on mount
  useEffect(() => {
    setMounted(true);
    // Load saved color from localStorage or use default
    const savedColor = localStorage.getItem('brand-color') || defaultColor;
    setSelectedColor(savedColor);
    setCustomColor(savedColor);

    // Apply the saved/default color to both light and dark modes
    applyColorToDocument(savedColor);
  }, [defaultColor, applyColorToDocument]); // Include dependencies

  const handleColorSelect = (color: string) => {
    setSelectedColor(color);
    setCustomColor(color);
    onColorChange?.(color);

    // Apply color changes to document and save to localStorage
    applyColorToDocument(color);
  };

  const handleCustomColorChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const color = e.target.value;
    setCustomColor(color);
    handleColorSelect(color);
  };

  const resetToDefault = () => {
    handleColorSelect(defaultColor);
  };

  // Render neutral state until mounted to prevent hydration mismatch
  if (!mounted) {
    return (
      <Button variant="outline" size="sm" className="flex items-center gap-2" disabled>
        <div
          className="w-4 h-4 rounded-full border border-border"
          style={{ backgroundColor: defaultColor }}
        />
        <Palette className="w-4 h-4" />
        <span className="hidden sm:inline">Brand Color</span>
      </Button>
    );
  }

  return (
    <div className={`relative ${className}`}>
      <Button
        variant="outline"
        size="sm"
        onClick={() => setIsOpen(!isOpen)}
        className="flex items-center gap-2"
      >
        <div
          className="w-4 h-4 rounded-full border border-border"
          style={{ backgroundColor: selectedColor }}
        />
        <Palette className="w-4 h-4" />
        <span className="hidden sm:inline">Brand Color</span>
      </Button>

      {isOpen && (
        <div className="absolute top-full right-0 mt-2 p-4 bg-background border border-border rounded-lg shadow-lg z-50 min-w-[280px]">
          <div className="space-y-4">
            <div>
              <h3 className="text-sm font-medium mb-2">Choose Brand Color</h3>
              <p className="text-xs text-muted-foreground mb-3">
                See your brand colors applied instantly across the entire site.
              </p>
            </div>

            {/* Preset Colors */}
            <div>
              <label className="text-xs font-medium text-muted-foreground mb-2 block">
                Preset Colors
              </label>
              <div className="grid grid-cols-6 gap-2">
                {PRESET_COLORS.map(color => (
                  <button
                    key={color}
                    onClick={() => handleColorSelect(color)}
                    className={`w-8 h-8 rounded-full border-2 transition-all hover:scale-110 ${
                      selectedColor === color
                        ? 'border-foreground ring-2 ring-offset-2 ring-offset-background ring-foreground/20'
                        : 'border-border hover:border-foreground/50'
                    }`}
                    style={{ backgroundColor: color }}
                    title={color}
                  />
                ))}
              </div>
            </div>

            {/* Custom Color */}
            <div>
              <label className="text-xs font-medium text-muted-foreground mb-2 block">
                Custom Color
              </label>
              <div className="flex gap-2">
                <input
                  type="color"
                  value={customColor}
                  onChange={handleCustomColorChange}
                  className="w-12 h-8 rounded border border-border cursor-pointer"
                />
                <input
                  type="text"
                  value={customColor}
                  onChange={e => {
                    setCustomColor(e.target.value);
                    if (isValidHexColor(e.target.value)) {
                      handleColorSelect(e.target.value);
                    }
                  }}
                  placeholder="#000000"
                  className="flex-1 px-2 py-1 text-sm border border-border rounded bg-background"
                />
              </div>
            </div>

            {/* Actions */}
            <div className="flex gap-2 pt-2 border-t border-border">
              <Button size="sm" variant="outline" onClick={resetToDefault} className="flex-1">
                Reset
              </Button>
              <Button size="sm" onClick={() => setIsOpen(false)} className="flex-1">
                Apply
              </Button>
            </div>

            <p className="text-xs text-muted-foreground">
              ðŸ’¡ Colors are automatically adjusted for proper contrast across light and dark themes
            </p>

            {/* Contrast Preview */}
            <div className="border border-border rounded p-2">
              <div className="text-xs font-medium mb-2">Contrast Preview:</div>
              <div className="grid grid-cols-2 gap-2 text-xs">
                <div
                  className="p-2 rounded text-center"
                  style={{
                    backgroundColor: selectedColor,
                    color:
                      hexToRgb(selectedColor) && getLuminance(hexToRgb(selectedColor)!) > 0.5
                        ? '#000000'
                        : '#ffffff',
                  }}
                >
                  Primary Button
                </div>
                <div
                  className="p-2 rounded text-center border"
                  style={{
                    borderColor: selectedColor,
                    color: selectedColor,
                    backgroundColor: 'transparent',
                  }}
                >
                  Outline Button
                </div>
              </div>
            </div>
          </div>
        </div>
      )}
    </div>
  );
}

// Helper functions
function hexToRgb(hex: string) {
  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  return result
    ? {
        r: parseInt(result[1], 16),
        g: parseInt(result[2], 16),
        b: parseInt(result[3], 16),
      }
    : null;
}

function getLuminance(rgb: { r: number; g: number; b: number }) {
  // Convert RGB to relative luminance using WCAG formula
  const rsRGB = rgb.r / 255;
  const gsRGB = rgb.g / 255;
  const bsRGB = rgb.b / 255;

  const r = rsRGB <= 0.03928 ? rsRGB / 12.92 : Math.pow((rsRGB + 0.055) / 1.055, 2.4);
  const g = gsRGB <= 0.03928 ? gsRGB / 12.92 : Math.pow((gsRGB + 0.055) / 1.055, 2.4);
  const b = bsRGB <= 0.03928 ? bsRGB / 12.92 : Math.pow((bsRGB + 0.055) / 1.055, 2.4);

  return 0.2126 * r + 0.7152 * g + 0.0722 * b;
}

function isValidHexColor(hex: string) {
  return /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/.test(hex);
}

export { ColorPicker };
export type { ColorPickerProps };
