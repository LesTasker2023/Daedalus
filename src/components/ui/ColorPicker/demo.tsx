'use client';

import { useState } from 'react';
import { ColorPicker } from '.';

export default function ColorPickerDemo() {
  const [color, setColor] = useState('#3B82F6');
  const [secondColor, setSecondColor] = useState('#EF4444');

  return (
    <div className="space-y-6 p-6 max-w-2xl mx-auto">
      <div className="space-y-2">
        <h1 className="text-2xl font-bold">ColorPicker Component</h1>
        <p className="text-muted-foreground">Color selection interface</p>
      </div>

      <div className="space-y-4">
        <div className="space-y-2">
          <h3 className="text-lg font-semibold">Basic Usage</h3>
          <div className="border rounded-lg p-4 bg-background">
            <ColorPicker defaultColor={color} onColorChange={setColor} />
            <p className="text-sm text-muted-foreground mt-2">Selected: {color}</p>
          </div>
        </div>

        <div className="space-y-2">
          <h3 className="text-lg font-semibold">Different Color</h3>
          <div className="border rounded-lg p-4 bg-background space-y-2">
            <ColorPicker defaultColor={secondColor} onColorChange={setSecondColor} />
            <p className="text-sm text-muted-foreground">Selected: {secondColor}</p>
          </div>
        </div>
      </div>
    </div>
  );
}
