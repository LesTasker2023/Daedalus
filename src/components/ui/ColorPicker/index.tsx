export * from './ColorPicker';
export type { ColorPickerProps } from './ColorPicker';
export { default as ColorPickerDemo } from './demo';
