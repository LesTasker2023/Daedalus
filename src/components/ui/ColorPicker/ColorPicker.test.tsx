import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import { ColorPicker } from './ColorPicker';

describe('ColorPicker', () => {
  const BasicColorPicker = ({ ...props }: { [key: string]: unknown }) => (
    <ColorPicker defaultColor="#18181b" aria-label="Test color picker" {...props} />
  );

  describe('Snapshots', () => {
    it('matches default snapshot', () => {
      const { container } = render(<BasicColorPicker />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches disabled state snapshot', () => {
      const { container } = render(<BasicColorPicker disabled />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches custom color snapshot', () => {
      const { container } = render(<BasicColorPicker defaultColor="#ff0000" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches with className snapshot', () => {
      const { container } = render(<BasicColorPicker className="custom-class" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Basic Functionality', () => {
    it('renders correctly', () => {
      render(<BasicColorPicker />);
      expect(screen.getByRole('button', { name: /test color picker/i })).toBeInTheDocument();
      expect(screen.getByText('Brand Color')).toBeInTheDocument();
    });

    it('displays correct default color', () => {
      render(<BasicColorPicker />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });
  });

  describe('States', () => {
    it('handles disabled state correctly', () => {
      render(<BasicColorPicker disabled />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeDisabled();
    });

    it('handles different default colors correctly', () => {
      render(<BasicColorPicker defaultColor="#ff0000" />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('handles no default color correctly', () => {
      render(<BasicColorPicker />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });
  });

  describe('Events', () => {
    it('handles onClick correctly', () => {
      render(<BasicColorPicker />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('handles onColorChange correctly', () => {
      const onColorChange = vi.fn();
      render(<BasicColorPicker onColorChange={onColorChange} />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });
  });

  describe('Props', () => {
    it('handles defaultColor prop correctly', () => {
      render(<BasicColorPicker defaultColor="#00ff00" />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('handles className prop correctly', () => {
      const { container } = render(<BasicColorPicker className="custom-class" />);
      const wrapper = container.firstChild as HTMLElement;
      expect(wrapper).toHaveClass('custom-class');
    });

    it('handles aria-label prop correctly', () => {
      render(<BasicColorPicker aria-label="Custom aria label" />);
      const button = screen.getByRole('button', { name: /custom aria label/i });
      expect(button).toBeInTheDocument();
    });
  });

  describe('Accessibility', () => {
    it('has proper ARIA attributes', () => {
      render(<BasicColorPicker />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toHaveAttribute('aria-label');
    });

    it('announces changes to screen readers', () => {
      render(<BasicColorPicker />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('respects reduced motion preferences', () => {
      render(<BasicColorPicker />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });
  });

  describe('Custom Styling and Props', () => {
    it('accepts custom className', () => {
      const { container } = render(<BasicColorPicker className="custom-class" />);
      const wrapper = container.firstChild as HTMLElement;
      expect(wrapper).toHaveClass('custom-class');
    });

    it('forwards aria-label correctly', () => {
      render(<BasicColorPicker aria-label="Custom label" />);
      const button = screen.getByRole('button', { name: /custom label/i });
      expect(button).toBeInTheDocument();
    });

    it('handles multiple props together', () => {
      render(<BasicColorPicker defaultColor="#ff0000" disabled className="test-class" />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeDisabled();
    });
  });

  describe('Edge Cases', () => {
    it('handles undefined defaultColor gracefully', () => {
      render(<BasicColorPicker defaultColor={undefined} />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('handles invalid hex colors gracefully', () => {
      render(<BasicColorPicker defaultColor="invalid-color" />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('handles empty string defaultColor', () => {
      render(<BasicColorPicker defaultColor="" />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('handles rapid prop changes', () => {
      const { rerender } = render(<BasicColorPicker defaultColor="#ff0000" />);
      rerender(<ColorPicker defaultColor="#00ff00" aria-label="Test color picker" />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('handles very long hex values', () => {
      render(<BasicColorPicker defaultColor="#ff0000ff" />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('handles short hex values', () => {
      render(<BasicColorPicker defaultColor="#f00" />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('handles component unmounting cleanly', () => {
      const { unmount } = render(<BasicColorPicker />);
      expect(() => unmount()).not.toThrow();
    });

    it('preserves functionality after remounting', () => {
      const { unmount } = render(<BasicColorPicker />);
      unmount();
      render(<BasicColorPicker />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('handles missing onColorChange gracefully', () => {
      render(<BasicColorPicker onColorChange={undefined} />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('handles null className gracefully', () => {
      render(<BasicColorPicker className={null as never} />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('maintains state during re-renders', () => {
      const { rerender } = render(<BasicColorPicker defaultColor="#ff0000" />);
      rerender(<ColorPicker defaultColor="#ff0000" aria-label="Test color picker" />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('handles edge case hex colors', () => {
      render(<BasicColorPicker defaultColor="#000000" />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });

    it('handles all white color', () => {
      render(<BasicColorPicker defaultColor="#ffffff" />);
      const button = screen.getByRole('button', { name: /test color picker/i });
      expect(button).toBeInTheDocument();
    });
  });
});
