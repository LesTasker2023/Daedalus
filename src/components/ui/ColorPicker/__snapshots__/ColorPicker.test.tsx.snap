// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ColorPicker > Snapshots > matches custom color snapshot 1`] = `
<div
  class="relative "
>
  <button
    aria-label="Test color picker"
    class="justify-center whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground [color:hsl(var(--surface-interactive))] [border-color:hsl(var(--surface-interactive))] h-8 rounded-md px-3 text-xs button flex items-center gap-2"
  >
    <div
      class="w-4 h-4 rounded-full border border-border"
      style="background-color: rgb(24, 24, 27);"
    />
    <svg
      aria-hidden="true"
      class="lucide lucide-palette w-4 h-4"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 22a1 1 0 0 1 0-20 10 9 0 0 1 10 9 5 5 0 0 1-5 5h-2.25a1.75 1.75 0 0 0-1.4 2.8l.3.4a1.75 1.75 0 0 1-1.4 2.8z"
      />
      <circle
        cx="13.5"
        cy="6.5"
        fill="currentColor"
        r=".5"
      />
      <circle
        cx="17.5"
        cy="10.5"
        fill="currentColor"
        r=".5"
      />
      <circle
        cx="6.5"
        cy="12.5"
        fill="currentColor"
        r=".5"
      />
      <circle
        cx="8.5"
        cy="7.5"
        fill="currentColor"
        r=".5"
      />
    </svg>
    <span
      class="hidden sm:inline"
    >
      Brand Color
    </span>
  </button>
</div>
`;

exports[`ColorPicker > Snapshots > matches default snapshot 1`] = `
<div
  class="relative "
>
  <button
    aria-label="Test color picker"
    class="justify-center whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground [color:hsl(var(--surface-interactive))] [border-color:hsl(var(--surface-interactive))] h-8 rounded-md px-3 text-xs button flex items-center gap-2"
  >
    <div
      class="w-4 h-4 rounded-full border border-border"
      style="background-color: rgb(24, 24, 27);"
    />
    <svg
      aria-hidden="true"
      class="lucide lucide-palette w-4 h-4"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 22a1 1 0 0 1 0-20 10 9 0 0 1 10 9 5 5 0 0 1-5 5h-2.25a1.75 1.75 0 0 0-1.4 2.8l.3.4a1.75 1.75 0 0 1-1.4 2.8z"
      />
      <circle
        cx="13.5"
        cy="6.5"
        fill="currentColor"
        r=".5"
      />
      <circle
        cx="17.5"
        cy="10.5"
        fill="currentColor"
        r=".5"
      />
      <circle
        cx="6.5"
        cy="12.5"
        fill="currentColor"
        r=".5"
      />
      <circle
        cx="8.5"
        cy="7.5"
        fill="currentColor"
        r=".5"
      />
    </svg>
    <span
      class="hidden sm:inline"
    >
      Brand Color
    </span>
  </button>
</div>
`;

exports[`ColorPicker > Snapshots > matches disabled state snapshot 1`] = `
<div
  class="relative "
>
  <button
    aria-label="Test color picker"
    class="justify-center whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground [color:hsl(var(--surface-interactive))] [border-color:hsl(var(--surface-interactive))] h-8 rounded-md px-3 text-xs button flex items-center gap-2"
    disabled=""
  >
    <div
      class="w-4 h-4 rounded-full border border-border"
      style="background-color: rgb(24, 24, 27);"
    />
    <svg
      aria-hidden="true"
      class="lucide lucide-palette w-4 h-4"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 22a1 1 0 0 1 0-20 10 9 0 0 1 10 9 5 5 0 0 1-5 5h-2.25a1.75 1.75 0 0 0-1.4 2.8l.3.4a1.75 1.75 0 0 1-1.4 2.8z"
      />
      <circle
        cx="13.5"
        cy="6.5"
        fill="currentColor"
        r=".5"
      />
      <circle
        cx="17.5"
        cy="10.5"
        fill="currentColor"
        r=".5"
      />
      <circle
        cx="6.5"
        cy="12.5"
        fill="currentColor"
        r=".5"
      />
      <circle
        cx="8.5"
        cy="7.5"
        fill="currentColor"
        r=".5"
      />
    </svg>
    <span
      class="hidden sm:inline"
    >
      Brand Color
    </span>
  </button>
</div>
`;

exports[`ColorPicker > Snapshots > matches with className snapshot 1`] = `
<div
  class="relative custom-class"
>
  <button
    aria-label="Test color picker"
    class="justify-center whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground [color:hsl(var(--surface-interactive))] [border-color:hsl(var(--surface-interactive))] h-8 rounded-md px-3 text-xs button flex items-center gap-2"
  >
    <div
      class="w-4 h-4 rounded-full border border-border"
      style="background-color: rgb(24, 24, 27);"
    />
    <svg
      aria-hidden="true"
      class="lucide lucide-palette w-4 h-4"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 22a1 1 0 0 1 0-20 10 9 0 0 1 10 9 5 5 0 0 1-5 5h-2.25a1.75 1.75 0 0 0-1.4 2.8l.3.4a1.75 1.75 0 0 1-1.4 2.8z"
      />
      <circle
        cx="13.5"
        cy="6.5"
        fill="currentColor"
        r=".5"
      />
      <circle
        cx="17.5"
        cy="10.5"
        fill="currentColor"
        r=".5"
      />
      <circle
        cx="6.5"
        cy="12.5"
        fill="currentColor"
        r=".5"
      />
      <circle
        cx="8.5"
        cy="7.5"
        fill="currentColor"
        r=".5"
      />
    </svg>
    <span
      class="hidden sm:inline"
    >
      Brand Color
    </span>
  </button>
</div>
`;
